acme/acme/messages.py

```diff
@@ -342,7 +355,7 @@ def from_data(cls, phone=None, email=None, external_account_binding=None, **kwar
 
     def __init__(self, **kwargs):
         """Note if the user provides a value for the `contact` member."""
-        if 'contact' in kwargs:
+        if 'contact' in kwargs and kwargs['contact'] is not None:
```

Comment:
Same comment about how I don't think `contact` should ever be done.
```suggestion
        if 'contact' in kwargs:
```

Reply:
As I said above, I am concerned about what happens in this new code if `None` is explicitly provided, leading to a situation where de-serialization results in `contact` containing `None` instead of a `tuple` and crashing when we try to iterate over it.

Reply:
Ah, nope, this was all chasing imaginary ghosts. Thank you for the catch.